src/Cms/Collection.php

```diff
@@ -158,13 +159,15 @@ public function findBy(string $attribute, $value)
 	 * Groups the items by a given field or callback. Returns a collection
 	 * with an item for each group and a collection for each group.
 	 *
-	 * @param string|Closure $field
+	 * @param string|\Closure $field
 	 * @param bool $caseInsensitive Ignore upper/lowercase for group names
-	 * @return \Kirby\Cms\Collection
+	 * @return self
 	 * @throws \Kirby\Exception\Exception
 	 */
-	public function group($field, bool $caseInsensitive = true)
-	{
+	public function group(
+		$field,
+		bool $caseInsensitive = true
+	): self {
```

Comment:
Should `Kirby\Toolkit\Collection::group()` really use `self` then or should that be switched to `static`?

Reply:
`Kirby\Toolkit\Collection::group()` currently uses `new Collection()`. I think the idea was that we e.g. wouldn't want to have a `Pages` object with the groups as each group is not a page but another collection. But it would indeed make sense to return a `Cms\Collection` at least from `Cms\Collection::group()`. That will be difficult to build though without adding a new argument for the class name to the `Toolkit\Collection::group()` method.

Reply:
Ok understand now the use case of not having it `static` all the way down to e.g. `Pages`.

`Kirby\Cms\Collection::group()` could return `Collection|BaseCollection`. Or we unwrap the BaseCollection from the `parent::group()` call and move the data into a CmsCollection. So it would always be `Kirby\Cms\Collection::group(): Kirby\Cms\Collection`

Reply:
Unwrapping sounds sensible. Then the return value will be consistent.

Reply:
I added a version, what do you think?